UnitTestingApp.js

```diff
@@ -52,24 +60,88 @@
       if (console.clear) console.clear();
     }
 
+    getLevelInfo() {
+      return this._levelInfo;
+    }
+
+    setLevelInfo(value) {
+      this._levelInfo = value;
+    }
+
+    resetTestCounters() {
+      this._nFailTests = 0;
+      this._nPassTests = 0;
+      this._nTests = 0;
+    }
+
     /**
      * Tests whether conditions pass or not
      * @param {Boolean | Function} condition - The condition to check
-     * @param {String} message - the message to display in the onsole
+     * @param {String} message - the message to display in the console (based on _levelInfo value)
      * @return {void}
      */
     assert(condition, message) {
-      if(!_enabled.get(this)) return;
-      if(this.isInGas !== this.runningInGas) return;
+      if (!_enabled.get(this)) return;
+      if (this.isInGas !== this.runningInGas) return;
+      this._nTests++;
       try {
         if ("function" === typeof condition) condition = condition();
-        if (condition) console.log(`✔ PASSED: ${message}`);
-        else console.log(`❌ FAILED: ${message}`);
-      } catch(err) {
-        console.log(`❌ FAILED: ${message} (${err})`);
+        if (condition) {
+          this._nPassTests++;
+          if (this._levelInfo > 0) { console.log(`✔ PASSED: ${message}`) };
+        }
+        else {
+          this._nFailTests++;
+          if (this._levelInfo > 0) { console.log(`❌ FAILED: ${message}`) };
+        }
+
+      } catch (err) {
+        this._nFailTests++;
+        if (this._levelInfo > 0) { console.log(`❌ FAILED: ${message} (${err})`) };
       }
     }
 
+    /**
+     * Tests whether fun result is equal to expected result or not
+     * @param {Function} fun - The function to evaluate
+     * @param {String} expectedResult - The expected result to validate
+     * @param {String} message - If present, then used as message to display in the console (based on _levelInfo value). 
+     *                           If message is not present, then in case the result is not equal, it shows the missmatch
+     *                           In the form of: "result != expectedResult". If the result is valid and message is not 
+     *                           provided, then it only indicates the test passed.
+     * @return {void}
+     */
+    assertEquals(fun, expectedResult, message = null) {
+      if (!_enabled.get(this)) return;
+      if (this.isInGas !== this.runningInGas) return;
+      this._nTests++;
+      let msg, result;
+
+      try {
+        if ("function" === typeof fun) {
```

Comment:
I know I do this in my code too, but we could user a ternary operator here you and I, it would take down one `if` :)

Reply:
About this suggestion, In my last version it is different for `assertEquals`, for `assert` you have the following line:
`if ("function" === typeof condition) condition = condition();`

which is correct, we don't have an else condition, so there is no way to use the ternary operator. If I am not correct, please advise. Thanks